frontend/src/lib/WorkflowParser.ts

```diff
@@ -231,30 +240,64 @@ export default class WorkflowParser {
     type ParamList = Array<KeyValue<S3Artifact>>;
     let inputArtifacts: ParamList = [];
     let outputArtifacts: ParamList = [];
-    if (
-      !nodeId ||
-      !workflow ||
-      !workflow.status ||
-      !workflow.status.nodes ||
-      !workflow.status.nodes[nodeId]
-    ) {
+
+    if (!workflow || !workflow.metadata || !workflow.status || !workflow.metadata.annotations)
       return { inputArtifacts, outputArtifacts };
-    }
 
-    const { inputs, outputs } = workflow.status.nodes[nodeId];
-    if (!!inputs && !!inputs.artifacts) {
-      inputArtifacts = inputs.artifacts.map(({ name, s3 }) => [name, s3]);
-    }
-    if (!!outputs && !!outputs.artifacts) {
-      outputArtifacts = outputs.artifacts.map(({ name, s3 }) => [name, s3]);
+    // Get the task name that corresponds to the nodeId
+    let taskName = "";
+    for (const taskRunId of Object.getOwnPropertyNames(workflow.status.taskRuns)) {
+      const taskRun = workflow.status.taskRuns[taskRunId];
+      if (taskRun.status && taskRun.status.podName === nodeId) {
+        taskName = taskRun.pipelineTaskName;
+      }
     }
-    return { inputArtifacts, outputArtifacts };
+
+    if (!taskName)
+      return { inputArtifacts, outputArtifacts };
+
+    const annotations = workflow.metadata.annotations
+    const rawInputArtifacts = annotations["tekton.dev/input_artifacts"] ? JSON.parse(annotations["tekton.dev/input_artifacts"]) : [];
+    const rawOutputArtifacts = annotations["tekton.dev/output_artifacts"] ? JSON.parse(annotations["tekton.dev/output_artifacts"]) : [];
+
+    let template =  {
+      endpoint: annotations['tekton.dev/artifact_endpoint'],
+      bucket: annotations['tekton.dev/artifact_bucket'],
+      insecure: annotations['tekton.dev/artifact_endpoint_scheme'] === 'http://',
+      accessKeySecret: {name: "mlpipeline-minio-artifact", key: "accesskey", optional: false},
+      secretKeySecret: {name: "mlpipeline-minio-artifact", key: "secretkey", optional: false},
+      key: `artifacts/${workflow.metadata.name || ""}`,
+    };
+
+    inputArtifacts = (rawInputArtifacts[taskName] || []).map((artifact : any) => {
+      return [ artifact.name, {
+        endpoint: template.endpoint,
+        bucket: template.bucket,
+        insecure: template.insecure,
+        accessKeySecret: template.accessKeySecret,
+        secretKeySecret: template.secretKeySecret,
+        key: `artifacts/${workflow.metadata.name}/${artifact.parent_task}/${artifact.name.substring(artifact.parent_task.length + 1)}.tgz`,
+      }]
+    })
+
+    outputArtifacts = (rawOutputArtifacts[taskName] || []).map((artifact : any) => {
+      return [ artifact.name, {
+        endpoint: template.endpoint,
+        bucket: template.bucket,
+        insecure: template.insecure,
+        accessKeySecret: template.accessKeySecret,
+        secretKeySecret: template.secretKeySecret,
+        key: artifact.key.replace("$PIPELINERUN", workflow.metadata.name || ""),
```

Comment:
Can we add a fallback here when artifact labels are not in the pod, we just ignore and skip the artifact div? Right now it will break if the pod wasn't ready or when it's pointing to a previous version of the pipeline.